hazelcast/listener.py

```diff
@@ -171,7 +170,10 @@ def callback(future):
                 return
 
             message.reset_next_frame()
-            self.handle_client_message(message, correlation_id)
+            try:
+                handler(message)
+            except SchemaNotFoundError as e:
+                self._fetch_schema_and_handle_again(e, handler, message)
```

Comment:
Do we need to recursively call this function? I think it is enough to try fetching the schema once. Let's say somehow schema is removed from the cluster, this event handler will be tried to be called forever.

Reply:
Schema cannot be removed from the cluster. Besides, you have to try recursively to deal with scenarios like
```
def event_handler(...)
  event_message = ...
  key = to_object(event_message["key_data"])
  value = to_object(event_message["value_data"])
```
where key and value are separate compact objects, and we don't have a schema for them

Reply:
Maybe later, the current logic both in the client and server is around fetching single schemas. We can do something like this maybe in 5.2